src/TBuffer.cpp

```diff
@@ -2430,11 +2430,11 @@ TBuffer TBuffer::copy(QPoint& P1, QPoint& P2)
     if ((x < 0) || (x >= static_cast<int>(buffer.at(y).size())) || (P2.x() < 0) || (P2.x() > static_cast<int>(buffer.at(y).size()))) {
         x = 0;
     }
-    int linkId, oldLinkId, id = 0;
+    int oldLinkId, id = 0;
```

Comment:
You still aren't initialising `oldLinkId` here, so it will be tested, holding a random value, against `linkId` in the `if` test! :neckbeard: 

:thinking:  Not sure whether:
```cpp
    if (linkId && !(linkId == oldLinkId)) {
```
is better than:
```cpp
    if (linkId && (linkId != oldLinkId)) {
```

Reply:
:man_facepalming:  wasn't aware that this doesn't give both the value of 0, but now that I read the link Vadi provided below I got it. (corrected it)

Thanks for all your help!